lib/_stream_readable.js

```diff
@@ -538,12 +538,12 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   }
   // This is a brutally ugly hack to make sure that our error handler
   // is attached before any userland ones.  NEVER DO THIS.
-  if (!dest._events || !dest._events.error)
+  if (!dest._events || !dest._events.has('error'))
     dest.on('error', onerror);
-  else if (Array.isArray(dest._events.error))
-    dest._events.error.unshift(onerror);
+  else if (Array.isArray(dest._events.get('error')))
+    dest._events.get('error').unshift(onerror);
   else
-    dest._events.error = [onerror, dest._events.error];
+    dest._events.set('error', [onerror, dest._events.get('error')]);
```

Comment:
Btw, there are other modules that are using _events directly. For example, https://github.com/mscdex/dicer/blob/master/lib/Dicer.js#L91